src/main/scala/mesosphere/marathon/tasks/TaskTracker.scala

```diff
@@ -5,6 +5,7 @@ import javax.inject.Inject
 import mesosphere.marathon.MarathonConf
```

Comment:
It is still full of race conditions, unfortunately.

But the steps of refactorings make it easier to eventually replace it (clearer scope in the functionality).